rust/src/client.rs

```diff
@@ -11,400 +11,324 @@ use tokio::{
-pub enum StakingLedgerArgs {
-    /// Query staking delegations by epoch
-    StakingDelegations(StakingDelegationsArgs),
-    /// Query staking delegations by public key and epoch
-    StakingPublicKey(StakingPublicKeyArgs),
+pub enum StakingLedger {
     /// Query staking ledger by hash
-    StakingLedgerHash(StakingLedgerHashArgs),
-    /// Query staking ledger by epoch
-    StakingLedgerEpoch(StakingLedgerEpochArgs),
-}
-
-#[derive(Args, Debug, Serialize, Deserialize)]
-#[command(author, version, about, long_about = None)]
-pub struct StakingLedgerEpochArgs {
-    /// Path to write the staking ledger [default: stdout]
-    #[arg(short, long)]
-    path: Option<PathBuf>,
-    /// Epoch number of the staking ledger
-    #[arg(short, long)]
-    epoch: u32,
-    /// Network
-    #[arg(short, long, default_value = "mainnet")]
-    network: String,
-}
-
-#[derive(Args, Debug, Serialize, Deserialize)]
-#[command(author, version, about, long_about = None)]
-pub struct StakingLedgerHashArgs {
-    /// Path to write the ledger [default: stdout]
-    #[arg(short, long)]
-    path: Option<PathBuf>,
-    /// Ledger hash corresponding to the staking ledger
-    #[arg(short, long)]
-    hash: String,
-    /// Network
-    #[arg(short, long, default_value = "mainnet")]
-    network: String,
-}
-
-#[derive(Args, Debug, Serialize, Deserialize)]
-#[command(author, version, about, long_about = None)]
-pub struct StakingPublicKeyArgs {
-    /// Epoch to aggregate staking delegations
-    #[arg(short, long)]
-    epoch: u32,
-    /// Account to aggregate staking delegations
-    #[arg(short = 'k', long)]
-    public_key: String,
-    /// Network for the staking ledger
-    #[arg(short, long, default_value = "mainnet")]
-    network: String,
-}
-
-#[derive(Args, Debug, Serialize, Deserialize)]
-#[command(author, version, about, long_about = None)]
-pub struct StakingDelegationsArgs {
-    /// Epoch to aggregate total delegations
-    #[arg(short, long)]
-    epoch: u32,
-    /// Network for the staking ledger
-    #[arg(short, long, default_value = "mainnet")]
-    network: String,
-    /// Path to write the aggregate delegations
-    #[arg(short, long)]
-    path: Option<PathBuf>,
-}
-
-#[derive(Parser, Debug, Serialize, Deserialize)]
+    Hash {
+        /// Ledger hash corresponding to the staking ledger
+        #[arg(long)]
+        hash: String,
+        /// Network
+        #[arg(long, default_value = "mainnet")]
+        network: String,
+        /// Path to write the staking ledger [default: stdout]
+        #[arg(long)]
+        path: Option<PathBuf>,
+    },
+    /// Query staking ledger at epoch
+    Epoch {
+        /// Epoch number of the staking ledger
+        #[arg(long)]
+        epoch: u32,
+        /// Network
+        #[arg(long, default_value = "mainnet")]
+        network: String,
+        /// Path to write the staking ledger [default: stdout]
+        #[arg(long)]
+        path: Option<PathBuf>,
+    },
+    /// Query staking ledger by ???
```

Comment:
@Isaac-DeFrain, can you recommend a better comment here?

Reply:
Query staking ledger delegations by epoch?